analytics/src/main/java/com/segment/analytics/internal/Utils.java

```diff
@@ -248,7 +248,11 @@ public static String assertNotNullOrEmpty(String text, @Nullable String name) {
     return Collections.unmodifiableList(new ArrayList<>(list));
   }
 
-  /** Creates a unique device id. */
+  /**
+   * Creates a unique device id. Suppresses `HardwareIds` lint warnings as we only rely on these
+   * APIs for metadata.
```

Comment:
This still is unclear to me why we can suppress this

Reply:
How can I make it clearer? This is what's required by the segment spec.

Reply:
hmm ok

Reply:
updated a bit more.